app/Http/Controllers/Api/V1/MicroSoftTeamController.php

```diff
@@ -132,24 +131,65 @@ public function getAccessToken(Request $request)
 	 */
     public function getAccessTokenViaCode(GetTokenViaCode $request)
     {
-            $accessToken = $this->microsoftTeamRepository->getTokenViaCode($request);
+        $accessToken = $this->microsoftTeamRepository->getTokenViaCode($request);
+
+        if ($accessToken && array_key_exists('access_token', $accessToken)) {
+            $request['token'] = $accessToken['access_token'];
+            $getSubmission = $this->microsoftTeamRepository->getSubmission($request);
 
-            if ($accessToken && array_key_exists('access_token', $accessToken)) {
-                $request['token'] = $accessToken['access_token'];
-                $getSubmission = $this->microsoftTeamRepository->getSubmission($request);
-                
+            if ($getSubmission && array_key_exists('status', $getSubmission)) {
                 return response([
                     'status_code' => 200,
                     'message' => 'Token fetched successfully.',
                     'access_token' => $accessToken['access_token'],
-                    'assignment_submission' => $getSubmission
+                    'assignment_submission' => $getSubmission,
+                    'refresh_token' => $accessToken['refresh_token']
                 ], 200);
             }
+
             return response([
                 'status_code' => 424,
-                'errors' => $accessToken['error'],
-                'message' => $accessToken['error_description']
-            ], 500);   
+                'errors' => $getSubmission['error'],
+            ], 500);
+        }
+        return response([
+            'status_code' => 424,
+            'errors' => $accessToken['error'],
+            'message' => $accessToken['error_description']
+        ], 500);
+    }
+
+    /**
+	 * Submit assignment
+	 *
+     * @bodyParam request contains classId, assignmentId, submissionId
+     * 
+     * @response {
+     *   "message": "Turned in successfully."
+     * }
+     *
+     * @response 500 {
+     *   "errors": [
+     *     "Failed to Turn in."
+     *   ]
+     * }
+     *
+     * @param SubmitAssignmentMst $request
+     * @return Response
+	 */
+    public function submitAssignment(SubmitAssignmentMst $request)
```

Comment:
Doc blocks need to update for the params required by SubmitAssignmentMst

Please consult following message by Waleed.
https://currikiworkspace.slack.com/archives/C01UQKSFDGB/p1668421603000099